tests/v2/validation/upgrade/kubernetes_test.go

```diff
@@ -106,4 +110,23 @@ func (u *UpgradeKubernetesTestSuite) testUpgradeSingleCluster(clusterName, versi
 		u.T().Logf("[%v]: Validating updated cluster's nodepools kubernetes versions", clusterName)
 		validateNodepoolVersions(u.T(), client, updatedCluster, version, !isCheckingCurrentCluster)
 	}
+
+	err = nodestat.IsNodeReady(client, clusterMeta.ID)
```

Comment:
imo we should add something like 
```
err = wait.WatchWait(result, checkFunc)
assert.NoError(t, err)
assert.Equal(t, clusterName, clusterResp.ObjectMeta.Name)
```
to also check that the cluster is in a ready state. In my testing, I found that all nodes ready != cluster ready.

Reply:
I can, but that leads to a point made in an earlier comment that it'd better to make that change for all of the other files. I don't necessarily have an issue with that, but it will make the scope of this PR larger.

Reply:
Further thinking of this, is this even necessary though? I mean, we have checks in the automation already to see if a cluster is active before we check for nodes status. Would that address your concern or am I misunderstanding?

Reply:
you're right, this is done earlier in this test.